### DISCUSSION TITLE
Not seeing "assigned" and "unassigned" IssuesEvents

### DISCUSSION THREAD

**Discussion by mscheper** [community/community#24885] [Apps, API and Webhooks]
<p>I’m using <a href="https://pygithub.readthedocs.io/" rel="nofollow noopener">PyGithub</a> to fetch events from one of our clients’ GitHub repos. Specifically, I’m calling <a href="https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository.get_events" rel="nofollow noopener">get_events</a> on the repository object, to <a href="https://developer.github.com/v3/activity/events/#list-issue-events-for-a-repository" rel="nofollow noopener">list recent issue events for the repo</a>.</p>
<p>This is working nicely, and it includes the <a href="https://developer.github.com/v3/activity/events/types/#issuesevent" rel="nofollow noopener">IssuesEvent</a>s that I expect to see, with opened, closed and reopened action types. But I’m <em>not</em> seeing ‘action’: ‘assigned’ or ‘action’: ‘unassigned’ IssuesEvents, even though I’ve recently changed Assignees for issues in GitHub.</p>
<p>I’m changing Assignees by going to an issue on the GitHub site, and clicking the gear icon alongside the Assignees. I’ve tried removing all Assignees, adding additional ones, etc. But IssuesEvents don’t seem to be being generated.</p>
<p>Any ideas for what I might be doing wrong?</p>
<p>(P.S: Please excuse the lack of formatting in this post. This forum seems to strip ‘code’ tags from posts, which I think is pretty silly.)</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/mscheper">@mscheper</a>,</p>
<p>Thank you for being here! Looking at PyGitHub’s documentation, the <code>get_events</code> call goes to the repo events list. Could you try this call instead: <a href="https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html?highlight=get_events#github.Repository.Repository.get_issues_events" rel="nofollow noopener">https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html?highlight=get_events#github.Repository.Repository.get_issues_events</a>.</p>
<p>That should work, it might otherwise be a limitation on the API side, although the documentation for the API says these events are listed, they are not available in the events endpoint PyGithub is reading from. You should be able to access these events via webhooks. I’ll ask for a sanity check on this from the API team and post any updates here.</p>
<p>Best,</p>
<p>Andrea</p>

**Comment by mscheper** [comment]
<p>G’day <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>Yep, using <a href="https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository.get_issues_events" rel="nofollow noopener">get_issues_events</a>, as you suggested, worked. I got <a href="https://pygithub.readthedocs.io/en/latest/github_objects/IssueEvent.html" rel="nofollow noopener">IssueEvent</a>s back instead of <a href="https://pygithub.readthedocs.io/en/latest/github_objects/Event.html" rel="nofollow noopener">Events</a>, which messes up my elegance a bit 😉, but it looks like all the data I need is there. Thanks for the answer, and for the welcome.</p>
